Userland/Libraries/LibWeb/Fetch/Infrastructure/HTTP/Headers.cpp

```diff
@@ -297,6 +297,42 @@ ErrorOr<Vector<Header>> HeaderList::sort_and_combine() const
     return headers;
 }
 
+// https://fetch.spec.whatwg.org/#header-list-extract-a-length
+Optional<u64> HeaderList::extract_length() const
+{
+    // 1. Let values be the result of getting, decoding, and splitting `Content-Length` from headers.
+    auto values_or_error = get_decode_and_split("Content-Length"sv.bytes());
+    if (values_or_error.is_error())
+        return {};
+    auto values = values_or_error.release_value();
+
+    // 2. If values is null, then return null.
+    if (!values.has_value())
+        return {};
+
+    // 3. Let candidateValue be null.
+    Optional<DeprecatedString> candidate_value;
+
+    // 4. For each value of values:
+    for (auto const& value : *values) {
+        // 1. If candidateValue is null, then set candidateValue to value.
+        if (!candidate_value.has_value()) {
+            candidate_value = value;
+        }
+        // 2. Otherwise, if value is not candidateValue, return failure.
```

Comment:
The fact that this function returns both `null` and `failure` tells me that something is not going to be happy with `Optional<T>` for both :)